cmd/contour/serve.go

```diff
@@ -217,32 +217,31 @@ func NewServer(log logrus.FieldLogger, ctx *serveContext) (*Server, error) {
 		MetricsBindAddress:     "0",
 		HealthProbeBindAddress: "0",
 		NewCache: ctrl_cache.BuilderWithOptions(ctrl_cache.Options{
-			// TransformByObject is a function that allows changing incoming objects before they are cached by the informer.
+			// ByObject is a function that allows changing incoming objects before they are cached by the informer.
 			// This is useful for saving memory by removing fields that are not needed by Contour.
-			TransformByObject: ctrl_cache.TransformByObject{
-				&corev1.Secret{}: func(obj interface{}) (interface{}, error) {
-					secret, ok := obj.(*corev1.Secret)
-					// TransformFunc should handle the tombstone of type cache.DeletedFinalStateUnknown
-					if !ok {
-						return obj, nil
-					}
-
-					// Do not touch Secrets that might be needed.
-					if secret.Type == corev1.SecretTypeTLS || secret.Type == corev1.SecretTypeOpaque {
-						return obj, nil
-					}
-
-					// Other types of Secrets will never be referred to, so we can remove all data.
-					// For example Secrets of type helm.sh/release.v1 can be quite large.
-					// Last-applied-configuration annotation might contain a copy of the complete data.
-					secret.Data = nil
-					secret.SetManagedFields(nil)
-					secret.SetAnnotations(nil)
-
-					// Returning error will avoid handlers from being called.
-					// This does not avoid caching.
-					return nil, fmt.Errorf("ignoring secret %s/%s of type %s", secret.Namespace, secret.Name, secret.Type)
```

Comment:
xref https://github.com/projectcontour/contour/pull/5366#issuecomment-1548660061 / https://github.com/projectcontour/contour/pull/5366#issuecomment-1550289190 -- agreed, seems like not returning an error is probably the way to go here

Reply:
When the nillified secret is propagated to the handler, there might be risk it will attempt to access `secret.Data` as a map? I guess [this](https://github.com/projectcontour/contour/blob/9605b63d5c0aa9f1802a642c5ce5b1c9e298dc5a/internal/dag/secret.go#L39) check saves it this time, but would it be safer to use empty map instead of nil, so that it would pass as a valid (empty) Secret?

```diff
- secret.Data = nil
+ secret.Data = map[string][]byte{}
```